File: Countries/js/4.js
// What is the shortest country name? Make sure your solution can handle ties.

let main = () => {

    const fs = require("fs");
    let buffer_arry = fs.readFileSync("Countries/countries.txt", "utf-8");
    let in_file = buffer_arry.split("\n");
    let short_obj = {};
    let is_first_word = true;
    let shortest_int = 0;

    // Function determines shortest words
    let get_lengths = (word) => {

        if(is_first_word == true) {
            shortest_int = word.length;
            short_obj[word] = word.length;
            is_first_word = false;
            return;
        };

        if(word.length < shortest_int) {
            shortest_int = word.length;
        };

        short_obj[word] = word.length;
    };

    // Function finds the words with the shortest recorded value
    let display_results = () => {
        for(let i = 0; i < Object.keys(short_obj).length; i++) {
            if(short_obj[Object.keys(short_obj)[i]] == shortest_int) {
                console.log(Object.keys(short_obj)[i]);
            };
        };
    };

    // Main function calls
    for(let i = 0; i < in_file.length; i++) {
        get_lengths(in_file[i]);
    };
    display_results();

};

main();

File: Countries/python/4.py
# What is the shortest country name? Make sure your solution can handle ties.
# Should "." count? Should "Micronesia (Federated States of)"?

def main():
    
    in_file = open("../countries.txt", "r")
    my_dict = {}
    for words in in_file:
        word_length_count = len(words)
        my_dict[words] = word_length_count
    tmp = min(my_dict.values())
    min_length = [key for key in my_dict if my_dict[key] == tmp]
    print(min_length)

if(__name__  == "__main__"):
    main()